src/targets/services/cleanOrphanAccounts.js

```diff
@@ -0,0 +1,97 @@
+const cozyFetch = require('../../lib/services/cozyFetch')
+
+const logMessagePrefix = level =>
+  `[cozy-collect][cleanOrphanAccounts][${level}]:`
+
+const output = level =>
+  function() {
+    console.log.apply(
+      console,
+      [logMessagePrefix(level)].concat(Object.values(arguments))
+    )
+  }
+
+const log = {
+  error: output('error'),
+  info: output('info')
+}
+
+const fetchAccounts = async () => {
+  log.info('Fetching accounts...')
+
+  const accounts = await cozyFetch(
+    'GET',
+    '/data/io.cozy.accounts/_all_docs?include_docs=true',
+    null,
+    true
+  )
+
+  log.info(`${accounts.length || 0} account(s) has been found.`)
+
+  return accounts
+}
+
+const fetchTriggers = async () => {
+  log.info('Fetching konnector triggers...')
+
+  const result = await cozyFetch(
+    'GET',
+    '/jobs/triggers?Worker=konnector',
+    null,
+    true
+  )
+
+  const hasTriggers = result && result.data && result.data.length
+  const triggers = result.data
+
+  if (hasTriggers) {
+    log.info(`${triggers.length || 0} konnector trigger(s) has been found.`)
+
+    return triggers
+  }
+
+  return []
+}
+
+const indexTriggersByAccounts = triggers =>
+  triggers.reduce((indexed, trigger) => {
+    const account =
+      trigger.attributes &&
+      trigger.attributes.message &&
+      trigger.attributes.message.account
+
+    return account ? { ...indexed, [account]: trigger } : indexed
+  }, {})
```

Comment:
Personnally I do like this pattern of building an object by building progressively building objects. I find that this is unnecessarily complicated. It is not very important here but if used regularly, it also puts unnecessarily pressure on the garbage collector by building/trashing a lot of objects.

A for loop would be sufficient I guess IMHO.